### ISSUE TITLE
Autocomplete DataSourceConfig Not getting passed down to AutoComplete component 

### ISSUE THREAD

**Issue by philipchmalts** [erikras/redux-form-material-ui#99]
```
<Field
          name="streetAddress"
          floatingLabelText="Address"
          dataSource={matches}
          onUpdateInput={getPredictions}
          component={AutoComplete}
          filter={AutoComplete.caseInsensitiveFilter}
          dataSourceConfig={{ text: 'description', value: 'place_id' }}
        />
```

When searching through all of the props From Field down to the actual autocomplete component dataSourceConfig is correct but on the actual AutoComplete component it is the default of
```
{
 text: 'text',
 value: 'value'
}
```

This forces me to have to change my output from external api to match that format resulting in an extra call to .map() . Works totally fine on normal AutoComplete component from MUI.

**Comment by robhadfield** [comment]
Yeah, I found this too - I map through to add the values as you say.

**Comment by hluedeke** [comment]
Try updating to v4.2.0

**Comment by erikras** [comment]
This should be fixed in v4.2.0.
